### DISCUSSION TITLE
How can I call emscripten function via DllImport in WebAssembly?

### DISCUSSION THREAD

**Discussion by Kation** [dotnet/runtime#81991] [Q&A] [answered]
```
[DllImport("__Internal", EntryPoint = "emscripten_run_script_int")]
public static extern int EM_ASM_INT(string code);
```
I don't know what dllName should be. It always show System.DllNotFoundException in browser console.

**Comment by maraf** [comment, marked as answer]
The `emscripten_run_script_int` is implemented in javascript and imported into the wasm module. In .NET 7 we don't support DllImport from wasm imports (dynamic linking of native libraries). You can bypass that by creating a stub that will be staticly linked.

```csharp
[LibraryImport("mylib", StringMarshalling = StringMarshalling.Utf16)]
public static partial int run_script_int(string code);
```

```c
#include <emscripten.h>

int run_script_int(char *code)
{
    return emscripten_run_script_int(code);
}
```

A whole example of linking native library can be found here https://github.com/dotnet/runtime/tree/main/src/mono/sample/wasm/browser-advanced.

**Anyway**, in case when you want to call into javascript from C#, it's better to use JSImport https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/import-export-interop (example in the same sample).

Let me know if you need more information.
